source/plugins/ruby/in_kube_perfinventory.rb

```diff
@@ -132,6 +133,18 @@ def enumerate(podList = nil)
         # Setting these to nil so that we dont hold memory until GC kicks in
         podInventory = nil
         nodeAllocatableRecords = nil
+
+        # Adding telemetry to send perfinventory telemetry every 5 minutes
+        timeDifference = (DateTime.now.to_time.to_i - @@perfTelemetryTimeTracker).abs
+        timeDifferenceInMinutes = timeDifference / 60
+        if (timeDifferenceInMinutes >= 5)
+          telemetryFlush = true
+        end
+
+        if telemetryFlush
+          ApplicationInsightsUtility.sendCustomEvent("KubePerfInventoryHeartBeatEvent", {})
```

Comment:
After emitting event, you would need to reset telemetryFlush = false, right? otherwise this will be emitted for every iteration.